tools/azsdk-cli/Azure.Sdk.Tools.Cli/Services/Languages/LanguageRepoService.cs

```diff
@@ -153,23 +176,117 @@ protected async Task<CLICheckResponse> ValidateChangelogCommonAsync(string packa
+                return new CLICheckResponse(1, "", $"Doc settings file not found at expected location: {settingsPath}");
             }
+
+            var command = "pwsh";
+            var args = new[] { 
+                "-File", scriptPath, 
+                "-SettingsPath", settingsPath,
+                "-ScanPaths", packagePath,
+            };
+
+            // Use a longer timeout for README validation - 10 minutes should be sufficient as it may need to install doc-warden
+            var timeout = TimeSpan.FromMilliseconds(600_000); // 10 minutes
+            var processResult = await _processHelper.Run(new(command, args, timeout: timeout, workingDirectory: packagePath), ct: default);
+
+            return CreateResponseFromProcessResult(processResult);
+        }
+        catch (Exception ex)
+        {
+            return new CLICheckResponse(1, "", $"Unhandled exception: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Common spelling check implementation that works for most Azure SDK languages.
+    /// Uses cspell directly to check spelling in the package directory.
+    /// </summary>
+    /// <param name="packagePath">Absolute path to the package directory</param>
+    /// <returns>CLI check response containing success/failure status and response message</returns>
+    protected async Task<CLICheckResponse> CheckSpellingCommonAsync(string packagePath)
+    {
+        try
+        {
+            if (!Directory.Exists(packagePath))
+            {
+                return new CLICheckResponse(1, "", $"Package path does not exist: {packagePath}");
+            }
+
+            // Find the SDK repository root by looking for common repository indicators
+            var packageRepoRoot = _gitHelper.DiscoverRepoRoot(packagePath);
+            if (string.IsNullOrEmpty(packageRepoRoot))
+            {
+                return new CLICheckResponse(1, "", $"Could not find repository root from package path: {packagePath}");
+            }
+
+            // Construct the path to the cspell config file
+            var cspellConfigPath = Path.Combine(packageRepoRoot, ".vscode", "cspell.json");
+            
+            if (!File.Exists(cspellConfigPath))
+            {
+                return new CLICheckResponse(1, "", $"Cspell config file not found at expected location: {cspellConfigPath}");
+            }
+
+            // Convert absolute path to relative path from repo root
+            var relativePath = Path.GetRelativePath(packageRepoRoot, packagePath);
+
+            // Build arguments for npx cspell lint
+            var args = new[] {
+                "cspell", "lint",
+                "--config", cspellConfigPath,
+                "--root", packageRepoRoot,
+                $"./{relativePath}/**"
+            };
+
+            var processResult = await _processHelper.Run(new("npx", args, workingDirectory: packageRepoRoot), ct: default);
```

Comment:
@timovv, for JS based commands, like this one, should we just recommend `npm install`ing this thing as a prerequisite or is 'npx' the right fit here? (I have no preference, just never occurred to me how often we use it)

Reply:
Perhaps we should make the helper do that, that's a pitfall just waiting to happen.

Reply:
For a bit @l0lawrence was randomly deleting code, but I think we should bring that one back to avoid issues like this one.

Reply:
@timovv, @l0lawrence: as usual, @chrisradek has really good code tucked away in his own PR that he refuses to share: https://github.com/Azure/azure-sdk-tools/pull/11628/files#diff-cfa866013827d7d2e5ff30443f91da9593664d510fc02005f3f84cd86cd28cb6

But once that's done, we should definitely kill our NPX related code in favor of that helper.

Reply:
@richardpark-msft You can already use the npx helper:
https://github.com/Azure/azure-sdk-tools/blob/b7f5662c46b340dc57421be587eb59e0a5f56723/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TypeSpec/TspTool.cs#L174-L190